### PR TITLE
Some PDW nodes

### PR DESCRIPTION
**Pull request by shashagit**

### CODE REVIEW

File: pdw/beta.hpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+class Beta : public BaseNode {
+	int P, vinit, vnext;
+	bool first;
+	int out;
+
+	void *P, *vinit, *vnext;
+	void *out;
+
+	SwitchTrue() : BaseNode(3, 1, beta) { first = TRUE;}
```
Comment (pranavk) [inline pdw/beta.hpp]: 3rd argument of BaseNode constructor should be a string. beta is undefined variable here, isn't it ?

Similarly, for all cases.

File: pdw/gamma.hpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ *
+ * This code is available to you under Apache License Version 2.0, January
+ * 2014. You can grab a copy of license from the same repository from where you
+ * fetched this code.
+ */
+
+#include "base_node.hpp"
+
+class Gamma : public BaseNode {
+	int P, vtrue, vfalse;
```
Comment (pranavk) [inline pdw/gamma.hpp]: Since P is only a predicate, why don't use bool for it, instead of int ?

**Comment by shashagit** [comment]
I have made the necessary changes.

File: pdw/gamma.hpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+class Gamma : public BaseNode {
+	bool P;
+	int vtrue, vfalse;
+	int out;
+
+	void *P, *vtrue, *vfalse;
+	void *out;
+
+	SwitchTrue() : BaseNode(3, 1, "gamma") {}
```
Comment (pranavk) [inline pdw/gamma.hpp]: I don't understand what you are trying to do here ?

Shouldn't it be Gamma() : BaseNode ( .... ?

**Comment by shashagit** [comment]
Sorry about that. Noob mistake. Take this :lollipop:

**Comment by pranavk** [comment]
Since I haven't yet merged your commits to master, you can squash these commits (about correcting nodes) into previous and upload a new patch set.

**Pull request closed without merging**